ports/atmel-samd/supervisor/port.c

```diff
@@ -496,6 +498,11 @@ void RTC_Handler(void) {
         // SAMD21 ticks are handled by EVSYS
         #ifdef SAM_D5X_E5X
         RTC->MODE0.INTENCLR.reg = RTC_MODE0_INTENCLR_CMP0;
+        // Check if we're sleeping
+        if (SAMD_ALARM_FLAG) {
```

Comment:
This is how `TimeAlarm` wakes from fake sleep. 

1. During setup of the time alarm, I make a call to [`port_interrupt_after_ticks()`](https://github.com/adafruit/circuitpython/blob/6811c5e88147701553fd1350f5be154e80ab0bd8/ports/atmel-samd/common-hal/alarm/time/TimeAlarm.c#L128) which sets the CMP0 interrupt. 
2. In the event of fake sleep, `main.py` eventually loops thru ['common_hal_alarm_pretending_deep_sleep()'](https://github.com/adafruit/circuitpython/blob/6811c5e88147701553fd1350f5be154e80ab0bd8/ports/atmel-samd/common-hal/alarm/__init__.c#L277). The flag is set the first time we hit 'common_hal_alarm_pretending_deep_sleep()'.
3. This allows the user to easily interrupt fake sleep via the REPL while also preserving the sleep timer behavior.

Right now, `SAMD_ALARM_FLAG` is utilizing the first backup register to store the flag. Do we want to avoid reading this backup register each time CMP0 fires and store the flag in flash instead?